src/platforms/java/common/performance/index.mdx

```diff
@@ -0,0 +1,103 @@
+
+You can learn more about the <PlatformIdentifier name="traces-sample-rate" /> and <PlatformIdentifier name="traces-sampler" /> options in <PlatformLink to="/performance/sampling/">Sampling Transactions</PlatformLink>.
+
+## Manual Instrumentation
+
+To manually instrument certain regions of your code, you can create a transaction to capture them.
+
+The following example creates a transaction for a scope that contains an expensive operation (for example, `processItem`), and sends the result to Sentry:
+
+```java
+import io.sentry.Sentry;
+import io.sentry.SentryTransaction;
+import io.sentry.SpanStatus;
+
+SentryTransaction transaction = Sentry.startTransaction("transaction name");
+transaction.setOperation("task");
+try {
+    processItem();
+} catch (Exception e) {
+    transaction.setThrowable(e);
+    transaction.setSpanStatus(SpanStatus.INTERNAL_ERROR)
+} finally {
+    transaction.finish();
+}
+```
+
+### Adding More Spans to the Transaction
+
+The next example contains the implementation of the hypothetical `processItem` function called from the code snippet in the previous section. Our SDK can determine if there is currently an open transaction and add all newly created spans as child operations to that transaction. Keep in mind that each individual span also needs to be manually finished; otherwise, spans will not show up in the transaction.
+
+In cases where you want to attach Spans to an already ongoing Transaction you can use `Sentry#getSpan`. This method will return a `SentryTransaction` in case there is a running Transaction or a `Span` in case there is already a running Span, otherwise it returns `null`.
+
+You can choose the value of `operation` and `description`.
+
+```java
+import io.sentry.ISpan;
+import io.sentry.Span;
+import io.sentry.SpanStatus;
+
+void processItem(Item item):
+    ISpan span = Sentry.getSpan();
+    Span innerSpan = span.startChild();
+    innerSpan.setOperation("task");
+    innerSpan.setDescription(item.getName());
+    try {
+        // omitted code
+    } catch (NotFoundException e) {
+        innerSpan.setThrowable(e);
+        innerSpan.setSpanStatus(SpanStatus.NOT_FOUND)
+    } finally {
+        innerSpan.finish();
+    }
+}
+```
+
+### Retrieving a Transaction
+
+In cases where you want to attach Spans to an already ongoing Transaction you can use `Sentry#getSpan`. This method will return a `SentryTransaction` in case there is a running Transaction or a `Span` in case there is already a running Span, otherwise it returns `null`.
+
+```java
+import io.sentry.ISpan;
+import io.sentry.Span;
+import io.sentry.SpanStatus;
+
+ISpan span = Sentry.getSpan();
```

Comment:
This won't really work with asynchronous code (this relies on TLS) so should we add a section to the docs about that?

What happens if I want to capture a span for an outgoing request that's happening on a background thread?

Reply:
Yes there is a single active transaction per `Hub`. I would imagine background thread should use a separate hub instance

Reply:
so if I want to have a transaction per App. screen, also being able to create spans and its children for background op (like outgoing requests), etc... having a Single Hub which is the case on Android, I don't see this as a problem unless I am missing something, is this described scenario possible to achieve with a single Hub?

Reply:
The limitation is that there can be single active transaction per hub

Reply:
the question still remains:

> is the described scenario above achievable with a single Hub?

the question is probably yes, but I'd like a confirmation.

Reply:
It depends. If the transaction in the parent thread is finished before the child thread's span is finished, child thread's span won't be sent together with transaction.

Reply:
I see, that makes sense, but for now, the support for performance on Android is "manual", so documenting this is just fine:

"finish your transaction only after your spans have finished".

Reply:
Thanks, updated!